Dockerfile

```diff
@@ -0,0 +1,26 @@
+## Development dockerfile for starfish, primarily
+## focused on the running of tests but the executables
+## are installed to ~starfish/.local/bin
+##
+## Examples:
+##
+##   docker build -t starfish .
```

Comment:
This is really great. How would you feel about adding a short explanation to each command? For those less familiar with docker, they might not know that these: (1) run help, (2) are equivalent to calling `starfish`, and (3) run verbose tests. 

Once this is built, what do you think about us pushing this up to quay or dockerhub and having it autobuild on new commits to master?

Reply:
More than happy to add the explanations. One question I had myself was whether you wanted the tests run by default or `starfish`. As for Dockerhub, by all means. Let me know to what extent you want me to create the org and/or repository.

Reply:
I think `starfish` would be preferable. @ttung what do you think?